File: LivrusVirtus/CPPBuilder6/CLX/dbExpress/fontes/ConAssuntos.cpp
//---------------------------------------------------------------------------

#include <clx.h>
#pragma hdrstop

#include "ConAssuntos.h"
#include "ConsultasSQL.h"
#include "Principal.h"
//---------------------------------------------------------------------------
#pragma link "Consultas"
#pragma resource "*.xfm"
TfrmConAssuntos *frmConAssuntos;
//---------------------------------------------------------------------------
__fastcall TfrmConAssuntos::TfrmConAssuntos(TComponent* Owner)
        : TfrmConsultas(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TfrmConAssuntos::FormClose(TObject *Sender,
      TCloseAction &Action)
{
frmPrincipal->Assuntos2->Enabled=true;
TfrmConsultas::FormClose(Sender, Action);
}
//---------------------------------------------------------------------------
void __fastcall TfrmConAssuntos::btnLimparClick(TObject *Sender)
{
LimparDados(ConsSQL->Assunto('S',"0",""));
}
//---------------------------------------------------------------------------
void __fastcall TfrmConAssuntos::btnPesquisarClick(TObject *Sender)
{
PesquisarDados(ConsSQL->Assunto('S',"", edtDescricao->Text),
"Assunto não encontrado!");
}
//---------------------------------------------------------------------------
void __fastcall TfrmConAssuntos::FormShow(TObject *Sender)
{
btnLimparClick(this);
}
//---------------------------------------------------------------------------

File: LivrusVirtus/Delphi7/CLX/dbExpress/fontes/ConAssuntos.pas
unit ConAssuntos;

interface

uses
  SysUtils, Types, Classes, Variants, QTypes, QGraphics, QControls, QForms, 
  QDialogs, QStdCtrls, Consultas, FMTBcd, DB, SqlExpr, QGrids, QDBGrids,
  QExtCtrls, Provider, DBClient;

type
  TfrmConAssuntos = class(TfrmConsultas)
    procedure FormClose(Sender: TObject; var Action: TCloseAction);
    procedure btnLimparClick(Sender: TObject);
    procedure btnPesquisarClick(Sender: TObject);
    procedure FormShow(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  frmConAssuntos: TfrmConAssuntos;

implementation

uses Principal, ConsultasSQL;

{$R *.xfm}

procedure TfrmConAssuntos.FormClose(Sender: TObject;
  var Action: TCloseAction);
begin
frmPrincipal.Assuntos2.Enabled:=true;
inherited;
end;

procedure TfrmConAssuntos.btnLimparClick(Sender: TObject);
begin
  inherited;
LimparDados(ConsSQL.Assunto('S','0',''));
end;

procedure TfrmConAssuntos.btnPesquisarClick(Sender: TObject);
begin
  inherited;
PesquisarDados(ConsSQL.Assunto('S','', edtDescricao.Text),
'Assunto não encontrado!');
end;

procedure TfrmConAssuntos.FormShow(Sender: TObject);
begin
  inherited;
btnLimparClick(self);
end;

end.
